labelme/labelDialog.py

```diff
@@ -90,19 +90,15 @@ def postProcess(self):
             self.edit.setText(self.edit.text().trimmed())
 
     def popUp(self, text=None, move=True):
-        if isinstance(text, str):
-            self.edit.setText(text)
-            self.edit.setSelection(0, len(text))
-            i = self.labelList.findItems(text, Qt.MatchFixedString)
-            if len(i) == 1:
-                self.labelList.setCurrentItem(i[0])
+        if text is None:
+            text = ''
+        self.edit.setText(text)
+        self.edit.setSelection(0, len(text))
+        items = self.labelList.findItems(text, Qt.MatchFixedString)
+        if items:
+            assert len(items) == 1
+            self.labelList.setCurrentItem(items[0])
         self.edit.setFocus(Qt.PopupFocusReason)
         if move:
             self.move(QCursor.pos())
         return self.edit.text() if self.exec_() else None
-
-    def keyPressEvent(self, e):
```

Comment:
Why did you remove the `keyPressEvent`?

Reply:
Do you really need this?
The label changes using arrow keys without this change in my env. It is only on macOS..?

Reply:
Could you re-check this?

Reply:
I'm on Ubuntu 16.04. In my case arrow keys only work if the listwidget has the focus, but we set the focus to the editbox. This keypress event allows moving the listwidget selection without first moving the focus to the listwidget.

Reply:
I've double checked it and without the keypress event I have to either press TAB 3 times or click on the listwidget before I can use the arrow keys for selecting a label from the list.

Reply:
I understood. But the implementation did not work on macOS, so I implemented it with another approach.

Reply:
For my understanding of python and pyQt: did the up and down key not work at all? Were the escape and enter key working for accepting and rejecting?

One thing I noticed when checking the differences is the call `else` case, my initial implementation:
```py
super().keyPressEvent(e)
```
your implementation:
```py
super(LabelQLineEdit, self).keyPressEvent(e)
```
Your implementation is more compatible as far as I understand from python docs, as `super()` was introduced in python 3 for convenience.

Reply:
> For my understanding of python and pyQt: did the up and down key not work at all? Were the escape and enter key working for accepting and rejecting?

Hmm, I'm not sure about your question, but all keys, up, down, escape, enter key work well;
up and down keys are for selecting label from list, escape and enter keys are for accepting or rejecting.

Reply:
> Your implementation is more compatible as far as I understand from python docs, as super() was introduced in python 3 for convenience.

Yes, my implementation is for compatibility on python2.